runtime/jcl/common/jclexception.c

```diff
@@ -97,6 +98,15 @@ getStackTraceIterator(J9VMThread * vmThread, void * voidUserData, J9ROMClass * r
 		return FALSE;
 	}
 
+	/* Check for extended modifiers on the ROM method */
+	if ((NULL != romMethod) && J9ROMMETHOD_HAS_EXTENDED_MODIFIERS(romMethod)) {
+		UDATA extraModifiers = getExtendedModifiersDataFromROMMethod(romMethod);
+		/* Skip the frame if @Hidden annotation is detected and -XX:+ShowHiddenFrames option has not been set */
+		if (J9_ARE_NO_BITS_SET(vm->runtimeFlags, J9_RUNTIME_SHOW_HIDDEN_FRAMES) && J9ROMMETHOD_HAS_LAMBDAFROM_HIDDEN_ANNOTATION(extraModifiers)) {
```

Comment:
This allows user code to add the annotation and hide itself in stack traces.  Does this match the RI's behaviour?

Reply:
tested with forcing annotation on non-lambda method, this matches RI's behaviour